### PR TITLE
Support alias use with hydration scripts

### LINKED ISSUE #2921: 🐛 BUG: Svelte components crash dev if imported with a shortened path
### What version of `astro` are you using?

0.25.3

### What package manager are you using?

yarn / npm / pnpm

### What operating system are you using?

Linux

### Describe the Bug

# How to reproduce

1. Create a project with a svelte integration
2. Place this in `tsconfig.json`:
```js
{
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "component:*": [
        "src/components/*"
      ]
    },
  }
}
```
3. Create a `components/Foo.svelte`, leave it empty
4. In `index.astro` write the following:
```astro
---
//import Foo from "component:Foo.svelte"
import Foo from "../components/Foo.svelte"
---
<Foo client:only/>
```
5. Run dev server
6. Notice everything working
7. Swap the commented/uncommented lines
8. Notice it failing with an error:
```
Cannot read property 'endsWith' of null

TypeError: Cannot read property 'endsWith' of null
    at ModuleGraph.resolveUrl (/sandbox/node_modules/vite/dist/node/chunks/dep-9c153816.js:53501:30)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async Object.resolve (file:///sandbox/node_modules/astro/dist/core/render/dev/index.js:96:45)
    at async Module.generateHydrateScript (/node_modules/astro/dist/runtime/server/hydration.js:61:152)
    at async Module.renderComponent (/node_modules/astro/dist/runtime/server/index.js:225:22)
    at async _render (/node_modules/astro/dist/runtime/server/index.js:43:11)
    at async renderAstroComponent (file:///sandbox/node_modules/astro/dist/runtime/server/index.js:389:20)
    at async renderPage (file:///sandbox/node_modules/astro/dist/runtime/server/index.js:346:20)
    at async render (file:///sandbox/node_modules/astro/dist/core/render/core.js:86:14)
    at async render (file:///sandbox/node_modules/astro/dist/core/render/dev/index.js:85:17)
```

# Issue

The error (likely) comes from these lines here in vite:
https://github.com/vitejs/vite/blob/4ffd82d2d4759bc3efa368bca54613aaa1f45acc/packages/vite/src/node/server/moduleGraph.ts#L222-L223

Because in our case, `url === "component:Foo.svelte"`, the parse for pathname will return null, erroring the subsequent check.

This is only a surface level analysis, and the root issue may be elsewhere. I do not know enough about how Astro interacts with vite to say.

### Link to Minimal Reproducible Example

https://codesandbox.io/s/bold-hill-yxjkmw?file=/src/pages/index.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- Fixes #2921
- Simplifies hydration id resolution in dev, now we just defer to `/@id`, an internal Vite resolution, unless it's already `/@fs`.

## Testing

Test added

## Docs

N/A, bug fix.

### CODE REVIEW

File: packages/astro/src/runtime/server/index.ts [hunk 1]
```diff
@@ -603,10 +603,7 @@ export async function renderHead(result: SSRResult): Promise<string> {
 			if ('data-astro-component-hydration' in script.props) {
 				needsHydrationStyles = true;
 			}
-			return renderElement('script', {
-				...script,
-				props: { ...script.props, 'astro-script': result._metadata.pathname + '/script-' + i },
-			});
+			return renderElement('script', script);
```
Comment (matthewp) [inline packages/astro/src/runtime/server/index.ts:606]: Note that this is not related to this change, just noticed that we were still adding these special attributes. They were used by the legacy build system but are not needed any more.

**Review by natemoo-re — APPROVED** [review]
Nice—looks great!

**Comment by florian-s-code** [comment]
I saw this PR got a revalidate flag so I thought I might as well comment here.
Please ignore if irrelevant but I was testing aliases on my side, more specifically on build. So I tried to write a test that would build the fixture created for this PR, but the build fails, the alias doesn't seem to work.

The start of the implementation that I couldn't finish: added a `describe` block but the whole `it` block inside is not yet modified:
https://github.com/florian-s-code/astro/commit/cbcbf98018e06af74698854ccf437c0d00318c0f
